test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/SemaphoreTest.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util.concurrent
+
+import java.util.concurrent.{Semaphore, TimeUnit}
+import org.junit.Assert._
+import org.junit.Test
+
+class SemaphoreTest {
+
+  @Test def `default constructor should be unfair`(): Unit = {
+    val sem = new Semaphore(1)
+    assertFalse(sem.isFair)
+  }
+
+  @Test def `constructor takes negative values`(): Unit = {
+    val sem = new Semaphore(-1)
+    assertEquals(-1, sem.availablePermits)
+  }
+
+
+  @Test def testDrainNegative(): Unit = {
+    val sem = new Semaphore(-1)
+    sem.drainPermits()
+    assertEquals(0, sem.availablePermits)
+  }
+
+  val acquireMethods: List[Semaphore => Unit] = List(
+    _.drainPermits(),
+    _.tryAcquire(),
+    _.tryAcquire(1),
+    _.tryAcquire(1, 1L, TimeUnit.DAYS),
+    _.tryAcquire(1L, TimeUnit.DAYS),
+  )
+
+  @Test def testAcquires(): Unit = {
```

Comment:
This should also test that `tryAcquire` fails if there are no permits. An maybe also test that release actually releases a permit.